internal/cfgplugins/interface.go

```diff
@@ -927,3 +927,120 @@ func NewAggregateInterface(t *testing.T, dut *ondatra.DUTDevice, b *gnmi.SetBatc
+	MagicIP  string // Per-port IP (e.g., "192.0.2.1")
+	MagicMAC string // Per-port MAC (e.g., "00:1A:2B:3C:4D:5E")
+}
+
+// StaticARPConfig holds all per-port static ARP entries.
+type StaticARPConfig struct {
+	Entries []StaticARPEntry
+}
+
+// StaticARPWithMagicUniversalIP configures static ARP and static routes per-port.
+func StaticARPWithMagicUniversalIP(t *testing.T, dut *ondatra.DUTDevice, cfg StaticARPConfig) {
+	t.Helper()
+	sb := &gnmi.SetBatch{}
+
+	// Group entries by MagicIP so each prefix can have multiple next-hops.
+	entriesByIP := make(map[string][]StaticARPEntry)
+	for _, entry := range cfg.Entries {
+		entriesByIP[entry.MagicIP] = append(entriesByIP[entry.MagicIP], entry)
+	}
+
+	for magicIP, entries := range entriesByIP {
+		// 1. Build all next-hops for this MagicIP.
+		nextHops := make(map[string]*oc.NetworkInstance_Protocol_Static_NextHop)
+		for i, entry := range entries {
+			port := dut.Port(t, entry.PortName)
+			nextHops[strconv.Itoa(i)] = &oc.NetworkInstance_Protocol_Static_NextHop{
+				Index: ygot.String(strconv.Itoa(i)),
+				InterfaceRef: &oc.NetworkInstance_Protocol_Static_NextHop_InterfaceRef{
+					Interface: ygot.String(port.Name()),
+				},
+			}
+		}
+
+		// 2. Define the static route with all built next-hops.
+		s := &oc.NetworkInstance_Protocol_Static{
+			Prefix:  ygot.String(magicIP + "/32"),
+			NextHop: nextHops,
+		}
+
+		// Add static route config to batch.
+		sp := gnmi.OC().
+			NetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+		gnmi.BatchUpdate(sb, sp.Static(magicIP+"/32").Config(), s)
+
+		// 3. Add static ARP entries separately for each port.
+		for _, entry := range entries {
+			port := dut.Port(t, entry.PortName)
+			gnmi.BatchUpdate(sb,
+				gnmi.OC().Interface(port.Name()).Config(),
+				configStaticArp(port.Name(), entry.MagicIP, entry.MagicMAC),
+			)
+
+			t.Logf("Configured ARP: port=%s, ip=%s, mac=%s",
+				entry.PortName, entry.MagicIP, entry.MagicMAC)
+		}
+
+		t.Logf("Configured static route for MagicIP %s with %d next-hops", magicIP, len(entries))
+	}
+
+	// 4. Commit everything in one batch.
+	sb.Set(t, dut)
+}
+
+// SecondaryIPEntry defines per-port dummy IP + ARP mapping for secondary IP config.
+type SecondaryIPEntry struct {
+	PortName      string           // DUT port name (e.g., "port2")
+	PortDummyAttr attrs.Attributes //  DUT dummy IP attributes
+	DummyIP       string           // OTG Dummy IPv4 address (e.g., "192.0.2.10")
+	MagicMAC      string           // MAC to use for static ARP (e.g., "00:1A:2B:3C:4D:FF")
+
+}
+
+// SecondaryIPConfig holds all per-port secondary IP configurations.
+type SecondaryIPConfig struct {
+	Entries []SecondaryIPEntry
+}
+
+// StaticARPWithSecondaryIP configures secondary IPs and static ARP for gRIBI compatibility
+func StaticARPWithSecondaryIP(t *testing.T, dut *ondatra.DUTDevice, cfg SecondaryIPConfig) {
```

Comment:
A batch object should be a parameter.  See [cfgplugins README](https://github.com/openconfig/featureprofiles/blob/main/internal/cfgplugins/README.md).

Reply:
we are not using batch object here.. could u pls explain the recommended changes..